lib/lunchbot_web/router.ex

```diff
@@ -50,7 +50,7 @@ defmodule LunchbotWeb.Router do
   # If your application does not have an admins-only section yet,
   # you can use Plug.BasicAuth to set up some basic authentication
   # as long as you are also using SSL (which you should anyway).
-  if Mix.env() in [:dev, :test] do
+  if Application.get_env(:lunchbot, :env) in [:dev, :test] do
```

Comment:
I think the phoenix generator spits out this code with `Mix.env()` referenced. We use releases on pineapple and don't have any problem with this usage of `Mix.env()`. I wonder if this is evaluated at compile time where mix is available? Basically... is this change really necessary?

Reply:
```
2022-07-14T13:53:07.000 [info]     ** (EXIT) an exception was raised:
2022-07-14T13:53:07.000 [info]         ** (UndefinedFunctionError) function Mix.env/0 is undefined (module Mix is not available)
```

https://community.fly.io/t/function-mix-env-0-is-undefined-module-mix-is-not-available/4181

Trying to deploy lunchbot to fly.io, and I'm running into the error described above.  And the answer above says that compiled releases do not include Mix.

Reply:
Could that error be referring to the `Mix.env()` usage in `Application.start`? I don't see anything in the error message that indicates that this specific usage of `Mix.env()` is the problem

Reply:
> I was assuming that the similar calls in the router were going to lead to the same error as well.

Yah makes sense, just wanted to provide perspective that I don't think this is the case.

> This is my first time tapping on an Elixir project. :)

It's weird having this distinction between compile-time and run-time. I think in this case, that the `Mix.env()` calls in `Router` will be called at compile time, where mix _is_ available :)

Reply:
This slightly verbose construction is also safe to use:

```elixir
if function_exported?(Mix, :env, 0) && Mix.env() == :dev do